assets/pages/new-contributions-landing/contributionsLandingInit.js

```diff
@@ -23,12 +25,38 @@ function initialisePaymentMethod(state: State, dispatch: Dispatch<Action>) {
   dispatch(updatePaymentMethod(paymentMethodToSelect));
 }
 
+function initialiseStripeCheckout(onPaymentAuthorisation, contributionType, currencyId, isTestUser, dispatch) {
+
+  setupStripeCheckout(onPaymentAuthorisation, contributionType, currencyId, isTestUser)
+    .then((handler: PaymentHandler) => dispatch(isPaymentReady(true, { [contributionType]: { Stripe: handler } })));
+}
+
+function initialisePaymentMethods(state: State, dispatch: Function) {
+  const { countryId } = state.common.internationalisation;
+  const { switches } = state.common.settings;
+  const { currencyId } = state.common.internationalisation;
+  const { isTestUser } = state.page.user;
+
+  const onPaymentAuthorisation = (paymentAuthorisation: PaymentAuthorisation) => {
+    dispatch(paymentWaiting(true));
+    dispatch(onThirdPartyPaymentAuthorised(paymentAuthorisation));
+  };
+
+  ['ONE_OFF', 'ANNUAL', 'MONTHLY'].forEach((contribType) => {
```

Comment:
I would refactor this into an inner loop over the payment methods:
```js
['ONE_OFF', 'ANNUAL', 'MONTHLY'].forEach(ct =>
  ['Stripe', 'PayPal', 'DirectDebit'].forEach(pm =>
    if (isValidPaymentMethod(ct, pm, switches, countryId))
      initPaymentMethod(...)
  )
)
```

Reply:
As we only want to initialise Stripe one off, and Stripe and PayPal recurring, writing out these specific commands rather than looping over all options.